frontend/scripts/react-components/profiles/line.component.jsx

```diff
@@ -173,35 +176,76 @@ class Line extends Component {
             .attr('d', line);
 
           if (!isSmallChart) {
-            pathContainers
+            const texts = pathContainers
               .selectAll('text')
               .data(d => [d])
-              .enter()
+              .enter();
+
+            texts
               .append('text')
               .attr('transform', d => {
                 const last = d.length - 1;
                 const { value } = d[last];
-                let newY = y(value) + 4;
-                if (newY + LINE_LABEL_HEIGHT > lastY) {
-                  newY = lastY - LINE_LABEL_HEIGHT;
+                let newNumberY = y(value) + 4;
+                if (newNumberY + LINE_LABEL_HEIGHT > lastNumberY) {
+                  newNumberY = lastNumberY - LINE_LABEL_HEIGHT - 1;
+                }
+                lastNumberY = newNumberY;
+                return `translate(${width + 6},${newNumberY})`;
+              })
+
+              .text(`${numLines - i}.`);
+
+            texts
```

Comment:
not sure why not make the link along with the number. But if we want to have link only on the text then we can also group those texts together to not duplicate `transform` function.
```
const texts = pathContainers
  .selectAll('text')
  .data(d => [d])
  .enter()
  .append('g')
  .attr('transform', d => {
    const last = d.length - 1;
    const { value } = d[last];
    let newNumberY = y(value) + 4;
    if (newNumberY + LINE_LABEL_HEIGHT > lastY) {
      newNumberY = lastY - LINE_LABEL_HEIGHT - 1;
    }
    lastY = newNumberY;
    return `translate(${width + 6},${newNumberY})`;
  });

texts.append('text').text(`${numLines - i}.`);

texts
  .append('text')
  .attr('transform', 'translate(16,0)')
  .attr('class', d => {
    if (typeof onLinkClick !== 'undefined' && d[0].nodeId && d[0].profileType) {
      return 'link';
    }
    return '';
  })
  .on('click', d => {
    if (typeof onLinkClick !== 'undefined' && d[0].nodeId && d[0].profileType) {
      onLinkClick(targetLink, {
        nodeId: d[0].nodeId,
        year
      });
    }
  })
  .text(d => `${capitalize(i18n(d[0].name))}`);
```

Reply:
the line under the number looked meh. Will try your approach, thanks